common/utils.js

```diff
@@ -767,20 +767,82 @@
+            }
+
+            // if resolverId is false or undefined OR any other values that are not allowed use the default
+            // default is to show the fully qualified resolveable link for permalink
+            return $window.location.origin + "/id/" + currCatalog + "/" + tuple.data.RID;
+        }
+
+        /**
+         * The following cases need to be handled for the resolverImplicitCatalog value:
+         *  - if null:                                      use current chaise path (include the version if none is present)
+         *  - if false:                                     /id/currCatalog/RID@version
+         *  - if undefined:                                 same as false
+         *  - if resolverImplicitCatalog == currCatalog:    /id/RID@version
+         *  - if resolverImplicitCatalog != currCatalog:    /id/currCatalog/RID@version
+         **/
+        function resolveVersionedPermalink(tuple, reference) {
+            var resolverId = chaiseConfig.resolverImplicitCatalog;
+            var currCatalog = reference.location.catalogId;
+            var snaptime = reference.table.schema.catalog.snaptime;
+            var version = reference.location.version;
+            var versionString = "@" + (version || snaptime);
+
+            // null or no RID
+            if (resolverId === null || !tuple.data || !tuple.data.RID) {
+                return $window.location.href.replace('#' + reference.location.catalog, '#' + currCatalog + versionString);
+            }
+
+            // if it's a number (isNaN tries to parse to integer before checking)
+            if (!isNaN(resolverId)) {
+                var catalog = (resolverId != currCatalog) ? currCatalog + "/" : "";
+                return $window.location.origin + "/id/" + catalog + tuple.data.RID + versionString;
+            }
+
+            // if resolverId is false or undefined OR any other values that are not allowed use the default
+            // default is to shwo the fully qualified resolveable link for permalink
+            return $window.location.origin + "/id/" + currCatalog + "/" + tuple.data.RID + versionString;
+        }
+
         return {
-            queryStringToJSON: queryStringToJSON,
+            appNamefromUrlPathname: appNamefromUrlPathname,
             appTagToURL: appTagToURL,
+            chaiseDeploymentPath: chaiseDeploymentPath,
             chaiseURItoErmrestURI: chaiseURItoErmrestURI,
+            createRedirectLinkFromPath: createRedirectLinkFromPath,
             fixedEncodeURIComponent: fixedEncodeURIComponent,
+            getQueryParams: getQueryParams,
+            isBrowserIE: isBrowserIE,
             parseURLFragment: parseURLFragment,
-            setOrigin: setOrigin,
             parsedFilterToERMrestFilter: parsedFilterToERMrestFilter,
+            queryStringToJSON: queryStringToJSON,
+            resolveLivePermalink: resolveLivePermalink,
+            resolveVersionedPermalink: resolveVersionedPermalink,
```

Comment:
I have 2 concerns with these 2 functions I added:
 1. They are almost identical to each other except the extra `versionString` calculation
 2. The properties I rely on in those functions (tuple and reference) are being passed as parameters. In both places that this is done, those parameters are passed from the `$rootScope`. These parameters could be removed in favor of the functions assuming they will always work off of the properties defined on the `$rootScope`. The current implementation allows for more future flexibility, but I'm not sure if that is even necessary to worry about.

Reply:
1. two functions:
If I remember correctly, the first implementation had more branching logic for "versioned" permalink. But based on this latest implementation, I don't see the point of having two separated functions. Based on what I understood the only difference is that in the versioned one, you're always  adding `versionString` to the result of the other function. 

If you want to have two functions without an optional parameter, you could call the "unversioned" one in the other function.

```javascript
function resolveVersionedPermalink(tuple, reference) {
  var version = reference.location.version;
  var versionString = "@" + (version || snaptime);
  return resolveLivePermalink(tuple, reference) + versionString;
}
```
It's up to you whether you want two functions. I don't have any strong opinion about it.

2. Since you're passing those variables, I would say leave it the way it is. Don't worry too much about it. You can change them later.